### PR TITLE
ensure bc clusters preserve the order

### PR DESCRIPTION
**Pull request by johrstrom**
I believe this is ready now. I believe `OodCore::Clusters`'s order is determined by the order the files are read which may be platform dependent.  This is currently breaking tests because our CI seems to be different than our laptops.

The main fix here is for `BatchConnect::App#cluster` to try to preserve the order of how things are defined in the `form.*` file, not the order they were read off disk which is how `OodCore::Clusters`' (and subsequently ood appkit) seem to behave.

Globs configurations like `o*` or just `*` for all clusters do not guarantee any order.

Test files have been updated because they had stubs at the wrong layer. Tests now read fixture files directly instead of stubbing at some other layer.

### CODE REVIEW

File: apps/dashboard/test/system/batch_connect_test.rb [hunk 1]
```diff
@@ -244,9 +244,11 @@ def setup
     Dir.mktmpdir('bc_cache_test') do |tmpdir|
       OodAppkit.stubs(:dataroot).returns(Pathname.new(tmpdir.to_s))
 
-      # puts "cache root is #{BatchConnect::Session.cache_root}"
+      a = BatchConnect::App.from_token('sys/bc_jupyter')
+      puts "#{a.send(:configured_clusters)} & #{a.send(:clusters).map(&:id).map(&:to_s)}"
```
Comment (johrstrom) [inline apps/dashboard/test/system/batch_connect_test.rb:248]:
```suggestion
```

File: apps/dashboard/test/system/batch_connect_test.rb [hunk 2]
```diff
@@ -244,9 +244,11 @@ def setup
...
 
       visit new_batch_connect_session_context_url('sys/bc_jupyter')
+      puts  find("##{bc_ele_id('cluster')}")['innerHTML']
```
Comment (johrstrom) [inline apps/dashboard/test/system/batch_connect_test.rb:249]:
```suggestion
```

File: apps/dashboard/app/models/batch_connect/app.rb [hunk 3]
```diff
@@ -152,13 +142,17 @@ def delete_cached_cluster?(cached_cluster)
...
 
-
     # The clusters that the batch connect app can use. It's a combination
     # of what the app is configured to use and what the user is allowed
-    # to use.
-    # @return [OodCore::Clusters] clusters available to the app user
+    # to use. If the app has clusters configured, it preserves the order
+    # when it can. glob configurations' order may be platform dependent.
+    #
+    # @return [Array<OodCore::Cluster>] clusters available to the app user
```
Comment (johrstrom) [inline apps/dashboard/app/models/batch_connect/app.rb:150]: Looks like the interface changes here - but not actually. The previous implementation returns the same, so it's actually the comment that's out of date.

```ruby
OodAppkit.clusters.select { |cluster| cluster_allowed(cluster) }
```

**Review by Oglopf — APPROVED** [review]
Looks good!

**Pull request merged**